### ISSUE TITLE
Uncaught TypeError: Cannot read property 'send' of undefined

### ISSUE THREAD

**Issue by Lianlizyj** [ashthespy/Atom-LaTeX#245]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.45.0 x64
**Electron**: 4.2.7
**OS**: Unknown Windows version
**Thrown From**: [atom-latex](https://github.com/ashthespy/Atom-LaTeX) package 0.9.1

### Stack Trace

Uncaught TypeError: Cannot read property 'send' of undefined

```
At C:\Users\Administrator\.atom\packages\atom-latex\lib\viewer.coffee:26

TypeError: Cannot read property 'send' of undefined
    at Viewer.module.exports.Viewer.wsHandler (/packages/atom-latex/lib/viewer.coffee:26:20)
    at /packages/atom-latex/lib/server.coffee:25:47)
    at WebSocket.emit (events.js:182:13)
    at Receiver.receiverOnMessage (/packages/atom-latex/node_modules/ws/lib/websocket.js:800:20)
    at Receiver.emit (events.js:182:13)
    at Receiver.dataMessage (/packages/atom-latex/node_modules/ws/lib/receiver.js:423:14)
    at Receiver.getData (/packages/atom-latex/node_modules/ws/lib/receiver.js:353:17)
    at Receiver.startLoop (/packages/atom-latex/node_modules/ws/lib/receiver.js:139:22)
    at Receiver._write (/packages/atom-latex/node_modules/ws/lib/receiver.js:74:10)
    at doWrite (_stream_writable.js:410:12)
    at writeOrBuffer (_stream_writable.js:394:5)
    at Receiver.Writable.write (_stream_writable.js:294:11)
    at Socket.socketOnData (/packages/atom-latex/node_modules/ws/lib/websocket.js:875:35)
    at Socket.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at Socket.Readable.push (_stream_readable.js:219:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

### Commands

```
     -0:42.2.0 latextools:build (input.hidden-input)
     -0:20 atom-latex:preview-tab (input.hidden-input)
```

### Non-Core Packages

```
atom-clock 0.1.18 
atom-latex 0.9.1 
Hydrogen 2.14.1 
language-fortran 2.1.10 
language-fortran 2.1.10 
language-latex 1.2.0 
latex 0.50.2 
latextools 0.8.5 
markdown-writer 2.11.10 
pdf-view 0.72.0 
```

**Comment by hallsville3** [comment]
I am having this issue as well on Atom 1.45.0 on macOS Catalina.

**Comment by ashthespy** [comment]
@hallsville3 @Lianlizyj Could you add some steps to reproduce this?

**Comment by jperrett256** [comment]
I get this particular issue whenever I select "Preview PDF" and then select "Preview PDF in new tab" (without closing the PDF preview window).

**Comment by stefaniecg** [comment]
How is this closed? Today is 4. March 2021 and i still have the error. 
When was this updated?

[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.45.0 x64
**Electron**: 4.2.7
**OS**: Mac OS X 10.13.6
**Thrown From**: [atom-latex](https://github.com/ashthespy/Atom-LaTeX) package 0.9.1


### Stack Trace

Uncaught TypeError: Cannot read property 'send' of undefined

```
At /Users/stefaniecg/.atom/packages/atom-latex/lib/viewer.coffee:26

TypeError: Cannot read property 'send' of undefined
    at Viewer.module.exports.Viewer.wsHandler (/packages/atom-latex/lib/viewer.coffee:26:20)
    at /packages/atom-latex/lib/server.coffee:25:47)
    at WebSocket.emit (events.js:182:13)
    at Receiver.receiverOnMessage (/packages/atom-latex/node_modules/ws/lib/websocket.js:825:20)
    at Receiver.emit (events.js:182:13)
    at Receiver.dataMessage (/packages/atom-latex/node_modules/ws/lib/receiver.js:437:14)
    at Receiver.getData (/packages/atom-latex/node_modules/ws/lib/receiver.js:367:17)
    at Receiver.startLoop (/packages/atom-latex/node_modules/ws/lib/receiver.js:143:22)
    at Receiver._write (/packages/atom-latex/node_modules/ws/lib/receiver.js:78:10)
    at doWrite (_stream_writable.js:410:12)
    at writeOrBuffer (_stream_writable.js:394:5)
    at Receiver.Writable.write (_stream_writable.js:294:11)
    at Socket.socketOnData (/packages/atom-latex/node_modules/ws/lib/websocket.js:900:35)
    at Socket.emit (events.js:182:13)
    at addChunk (_stream_readable.js:283:12)
    at readableAddChunk (_stream_readable.js:264:11)
    at Socket.Readable.push (_stream_readable.js:219:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

### Commands

```
     -1:09.9.0 atom-latex:clean (input.hidden-input)
     -1:01.8.0 tree-view:remove (span.name.icon.icon-file-zip)
     -0:46.5.0 atom-latex:preview-tab (div.tool-panel.tree-view)
  2x -0:37.7.0 tabs:split-right (div.title)
     -0:15.3.0 tabs:split-left (div.title)
```

### Non-Core Packages

```
atom-folding 0.0.9 
atom-latex 0.9.1 
busy-signal 2.0.1 
highlight-selected 0.17.0 
Hydrogen 2.14.7 
intentions 1.1.5 
language-latex 1.2.0 
latex 0.50.2 
latex-folding 1.5.0 
linter 3.3.0 
linter-spell 0.15.0 
linter-spell-latex 0.11.0 
linter-ui-default 3.0.2 
markdown-folding 0.3.1 
markdown-pdf 2.3.3 
markdown-preview-plus 3.11.5 
markdown-writer 2.11.11 
pdfjs-viewer 1.0.0 
platformio-ide-terminal 2.10.1 
quick-highlight 0.13.0 
readonly-utils 0.1.0 
script 3.29.3 
thesaurus 0.3.0 
view-mode 0.1.2 
```

**Comment by ashthespy** [comment]
> How is this closed? Today is 4. March 2021 and i still have the error.

Greetings, this was solved by 4e32284. Unfortunately I've been held up with other work, so haven't got enough time right now to publish a new version. You can always copy paste the code from the commit.. :-)
